client/src/app/pages/assessment/components/assessment-wizard/assessment-wizard.tsx

```diff
@@ -439,11 +440,20 @@ export const AssessmentWizard: React.FC<AssessmentWizardProps> = ({
     }
   };
 
-  const haveAnyQuestionBeenAnswered = () => {
-    const questions = values[QUESTIONS_KEY];
-    return Object.values(questions).some(
-      (answer) => answer !== null && answer !== ""
+  const isAssessmentChanged = () => {
+    const questionsChanged = Object.entries(values[QUESTIONS_KEY]).some(
+      ([name, answer]) => initialQuestions[name] !== answer
     );
+    const stakeholdersChanged =
+      !values.stakeholders.every(({ id, name }) =>
+        initialStakeholders.find((it) => it.id === id && it.name === name)
+      ) ||
+      initialStakeholders.length !== values.stakeholders.length ||
+      !values.stakeholderGroups.every(({ id, name }) =>
+        initialStakeholderGroups.find((it) => it.id === id && it.name === name)
+      ) ||
+      initialStakeholderGroups.length !== values.stakeholderGroups.length;
```

Comment:
Nit-pick: I'd put the length checks first since they're faster than the iterations.  I wouldn't expect it to be an actual problem with the potential sizes of these arrays, just a nit-pick.